x-pack/plugins/searchprofiler/public/application/components/profile_query_editor/profile_query_editor.tsx

```diff
@@ -86,6 +90,34 @@ export const ProfileQueryEditor = memo(() => {
   );
   const licenseEnabled = getLicenseStatus().valid;
 
+  useEffect(() => {
+    setHasIndices(indicesData?.ok ? indicesData?.hasIndices : false);
+  }, [indicesData]);
```

Comment:
I think we can remove the complexity of the useEffect and the useState by leveraging the states returned from the `useIndices` hook. 

```typescript
  const { data: indicesData, isLoading, error } = useIndices();
  ...
  const hasIndices = (isLoading || error) ? false : indicesData?.hasIndices;

```

Reply:
Changed!